txnsync/encodedgroupsunmarshalers.go

```diff
@@ -0,0 +1,644 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package txnsync
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/transactions"
+)
+
+var errDataMissing = errors.New("failed to decode: data missing")
+
+// the nextSlice definition - copy the next slice and slide the src window.
+func nextSlice(src *[]byte, dst []byte, size int) error {
```

Comment:
It would be much more error prone to leave it to the caller; The pattern 
```golang
	err = stub.BitmaskRekeyTo.iterate(int(stub.TotalTransactionsCount), len(stub.RekeyTo)/crypto.DigestSize, func(i int, index int) error {
		return nextSlice(&stub.RekeyTo, signedTxns[i].Txn.RekeyTo[:], crypto.DigestSize)
	})
```
is a good way to ensure that as long as we've followed the parameters correctly, we're good to go. By moving this logic to each one of the callers, we're making the code less coherent.

Reply:
Actually, no need to move the logic to the callers. The callers always fill the `dst` slice completely.